packages/eslint-plugin/tests/rules/no-useless-template-literals.test.ts

```diff
@@ -358,14 +362,92 @@ ruleTester.run('no-useless-template-literals', rule, {
     },
 
     {
-      code: "`a${'b'}`;",
-      output: '`ab`;',
+      code: "`use${'less'}`;",
+      output: '`useless`;',
       errors: [
         {
           messageId: 'noUselessTemplateLiteral',
           line: 1,
-          column: 5,
-          endColumn: 8,
+        },
+      ],
+    },
+
+    {
+      code: '`use${`less`}`;',
+      output: '`useless`;',
+      errors: [
+        {
+          messageId: 'noUselessTemplateLiteral',
+          line: 1,
+        },
+      ],
+    },
+
+    {
+      code: `
+declare const nested: string, interpolation: string;
+\`use\${\`less\${nested}\${interpolation}\`}\`;
+      `,
+      output: `
+declare const nested: string, interpolation: string;
+\`useless\${nested}\${interpolation}\`;
+      `,
+      errors: [
+        {
+          messageId: 'noUselessTemplateLiteral',
+        },
+      ],
+    },
+
+    {
+      code: noFormat`
+\`u\${
+  // hopefully this comment is not needed.
```

Comment:
[Non-Actionable] Oh, right, this reminds me I'd wanted to file a low-priority followup bug about the comments being removed. Thanks 😄 #8609